packages/serve/src/index.ts

```diff
@@ -1,39 +1,10 @@
-import { devServer } from 'webpack-dev-server/bin/cli-flags';
-import WebpackCLI from 'webpack-cli';
-import logger from 'webpack-cli/lib/utils/logger';
-import startDevServer from './startDevServer';
-
-/**
- *
- * Creates a webpack compiler and runs the devServer
- *
- * @param {String[]} args - args processed from the CLI
- * @returns {Function} invokes the devServer API
- */
-export default function serve(...args: string[]): void {
-    const cli = new WebpackCLI();
-    const core = cli.getCoreFlags();
-
-    const parsedDevServerArgs = cli.argParser(devServer, args, true);
-    const devServerArgs = parsedDevServerArgs.opts;
-    const parsedWebpackArgs = cli.argParser(core, parsedDevServerArgs.unknownArgs, true, process.title);
-    const webpackArgs = parsedWebpackArgs.opts;
-
-    // pass along the 'hot' argument to the dev server if it exists
-    if (webpackArgs && webpackArgs.hot !== undefined) {
-        devServerArgs['hot'] = webpackArgs.hot;
-    }
-
-    if (parsedWebpackArgs.unknownArgs.length > 0) {
-        parsedWebpackArgs.unknownArgs
-            .filter((e) => e)
-            .forEach((unknown) => {
-                logger.warn('Unknown argument:', unknown);
-            });
-        return;
-    }
-
-    cli.getCompiler(webpackArgs, core).then((compiler): void => {
-        startDevServer(compiler, devServerArgs);
-    });
+// Test dev-server availability before importing it
+try {
+    // eslint-disable-next-line node/no-extraneous-require
+    require.resolve('webpack-dev-server');
+} catch (error) {
+    throw new Error(`You need to install 'webpack-dev-server' for running 'webpack serve'.\n${error}`);
 }
+
+import serve from './serve';
+export default serve;
```

Comment:
Before import can not be something, it is work because we use ts transpiler, but it is bad practice

Reply:
Yeah, I wasn't happy with the solution either, but we should not import from 'webpack-dev-server', before checking it's existance and `serve.ts`( original index file) is dependent on that. So, can you point me where do you like to have this check then?

Reply:
Just put it in one file and do it inside `server`, using here is not safe, because it will output error for all developers who uses webpack-cli

Reply:
Okay, just be clear, is this what you are suggesting in the single file?
```js
import something from 'webpack-dev-server';
 // Inside serve method...
try {
   require.resolve('webpack-dev-server');
} catch (err){
// Handle err
}
//....
```

Reply:
```js
let devServer;

try {
   devServer = require('webpack-dev-server');
} catch (err){
  // Handle err
}

devServer.listen()
```